libass/ass_directwrite.c

```diff
@@ -530,19 +543,28 @@ static int map_width(enum DWRITE_FONT_STRETCH stretch)
 }
 
 static char *get_utf8_name(IDWriteLocalizedStrings *names, int k) {
-    wchar_t temp_name[NAME_MAX_LENGTH];
-    HRESULT hr = IDWriteLocalizedStrings_GetString(names, k,
-                                                   temp_name,
-                                                   NAME_MAX_LENGTH);
-    if (FAILED(hr))
+    UINT32 length;
+    HRESULT hr = IDWriteLocalizedStrings_GetStringLength(names, k, &length);
+    if (FAILED(hr) || !++length)
         return NULL;
 
-    temp_name[NAME_MAX_LENGTH-1] = 0;
+    wchar_t *temp_name = (wchar_t *) calloc(length, sizeof(wchar_t));
```

Comment:
I use `calloc` here not to zero the memory but to avoid argument overflow. If I were to multiply `length * sizeof(wchar_t)`, I’d need to add overflow checks.

Reply:
I guess I have an overflow check anyway (for `length + 1`), so I’ve changed this to an explicit check and `malloc`.